File: generators/narniaDwarfs.py
__all__ = ['narniaDwarfs']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm3', 'nm7', 'nm9', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm8').get('length'))|Js(0.0)))
                var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm9').get('length'))|Js(0.0)))
                var.put('names', (((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm8').get(var.get('rnd3')))+var.get('nm9').get(var.get('rnd4'))))
            else:
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                if (var.get('rnd3')<Js(9.0)):
                    var.put('names', (((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+Js('e')))
                else:
                    var.put('lng', ((var.get('Math').callprop('random')*Js(2.0))|Js(0.0)))
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                    var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                    if PyJsStrictEq(var.get('lng'),Js(0.0)):
                        var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm5').get(var.get('rnd5'))))
                    else:
                        var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                        var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                        var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd6')))+var.get('nm7').get(var.get('rnd7')))+var.get('nm5').get(var.get('rnd5'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('br'), Js('cr'), Js('d'), Js('dr'), Js('g'), Js('gr'), Js('p'), Js('r'), Js('th'), Js('tr'), Js('thr')]))
var.put('nm2', Js([Js('a'), Js('i'), Js('e'), Js('o'), Js('u')]))
var.put('nm3', Js([Js('bl'), Js('bbl'), Js('ckl'), Js('dl'), Js('ddl'), Js('ffl'), Js('fl'), Js('gl'), Js('ggl'), Js('d'), Js('g'), Js('gg'), Js('k'), Js('l'), Js('ll'), Js('n'), Js('v'), Js('z'), Js('sn'), Js('gn'), Js('gd'), None, Js('d'), Js('g'), Js('gg'), Js('k'), Js('l'), Js('ll'), Js('n'), Js('v'), Js('z'), Js('sn'), Js('gn'), Js('gd'), Js('d'), Js('g'), Js('gg'), Js('k'), Js('l'), Js('ll'), Js('n'), Js('v'), Js('z'), Js('sn'), Js('gn'), Js('gd'), Js('dsh'), Js('lnd'), Js('lng'), Js('mdr'), Js('mgl'), Js('mpk'), Js('ndr'), Js('ngd'), Js('ngb'), Js('nkr'), Js('rgl'), Js('rnb'), Js('sdr'), Js('sgl')]))
var.put('nm4', Js([Js('br'), Js('bl'), Js('dr'), Js('dn'), Js('fn'), Js('fd'), Js('fl'), Js('ld'), Js('ln'), Js('lm'), Js('mb'), Js('md'), Js('ml'), Js('nl'), Js('nd'), Js('rb'), Js('rl'), Js('rbr'), Js('th')]))
var.put('nm5', Js([Js('d'), Js('g'), Js('k'), Js('lk'), Js('ld'), Js('n'), Js('nk'), Js('nks'), Js('t')]))
var.put('nm6', Js([Js('b'), Js('cl'), Js('d'), Js('f'), Js('fl'), Js('g'), Js('gl'), Js('m'), Js('n'), Js('w')]))
var.put('nm7', Js([Js('a'), Js('e'), Js('i'), Js('o')]))
var.put('nm8', Js([Js('bb'), Js('bl'), Js('ch'), Js('fl'), Js('fs'), Js('fn'), Js('fm'), Js('gs'), Js('ks'), Js('ln'), Js('ls'), Js('lp'), Js('ms'), Js('mn'), Js('nl'), Js('nt'), Js('ns'), Js('nf'), Js('pf'), Js('ps'), Js('sl'), Js('sn'), Js('sm')]))
var.put('nm9', Js([Js('ie'), Js('ey'), Js('ee'), Js('i'), Js('e'), Js('iy')]))
pass
pass


# Add lib to the module scope
narniaDwarfs = var.to_python()

File: js/narniaDwarfs.js
var nm1 = ["br","cr","d","dr","g","gr","p","r","th","tr","thr"];
var nm2 = ["a","i","e","o","u"];
var nm3 = ["bl","bbl","ckl","dl","ddl","ffl","fl","gl","ggl","d","g","gg","k","l","ll","n","v","z","sn","gn","gd",,"d","g","gg","k","l","ll","n","v","z","sn","gn","gd","d","g","gg","k","l","ll","n","v","z","sn","gn","gd","dsh","lnd","lng","mdr","mgl","mpk","ndr","ngd","ngb","nkr","rgl","rnb","sdr","sgl"];
var nm4 = ["br","bl","dr","dn","fn","fd","fl","ld","ln","lm","mb","md","ml","nl","nd","rb","rl","rbr","th"];
var nm5 = ["d","g","k","lk","ld","n","nk","nks","t"];

var nm6 = ["b","cl","d","f","fl","g","gl","m","n","w"];
var nm7 = ["a","e","i","o"];
var nm8 = ["bb","bl","ch","fl","fs","fn","fm","gs","ks","ln","ls","lp","ms","mn","nl","nt","ns","nf","pf","ps","sl","sn","sm"];
var nm9 = ["ie","ey","ee","i","e","iy"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");

	for(i = 0; i < 10; i++){
		if(tp === 1){
			rnd = Math.random() * nm6.length | 0;
			rnd2 = Math.random() * nm7.length | 0
			rnd3 = Math.random() * nm8.length | 0;
			rnd4 = Math.random() * nm9.length | 0;
			names = nm6[rnd] + nm7[rnd2] + nm8[rnd3] + nm9[rnd4];
		}else{
			rnd = Math.random() * nm1.length | 0;
			rnd2 = Math.random() * nm2.length | 0
			rnd3 = Math.random() * nm3.length | 0;
			if(rnd3 < 9){
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + "e";
			}else{
				lng = Math.random() * 2 | 0;
				rnd4 = Math.random() * nm2.length | 0;
				rnd5 = Math.random() * nm5.length | 0;
				if(lng === 0){
					names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm5[rnd5];
				}else{
					rnd6 = Math.random() * nm4.length | 0;
					rnd7 = Math.random() * nm7.length | 0;
					names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm4[rnd6] + nm7[rnd7] + nm5[rnd5];
				}
			}
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}